bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSink.java

```diff
@@ -105,6 +108,7 @@ private InputStream getPCMStreamFromMp3Stream(InputStream input) {
      * @throws UnknownHostException
      * @throws IOException
      */
+    @SuppressWarnings("null")
     private void connectIfNeeded() throws IOException {
         if (clientSocket == null || !clientSocket.isConnected() || clientSocket.isClosed()) {
```

Comment:
You could store `clientSocket` to a local variable and do the null check on that. Then, you can remove the suppression. Same for below with the bridge.

Reply:
Yes thank you ! It is indeed more elegant.

I just pushed one correction (after finding some little legacy bug about deregistering a device listener), and clear some old compilation warnings. I hope it's OK to do this in the same pull request.

And I also made a little improvement : I just realize that the binding could handle the loading of the module remotely, instead of letting the user doing it.
I didn't squash my commit to let you review the diff. But if you think I should have, please tell me.
About this : I suppose the merge system auto squash commits before fast forwarding, so I don't have to do it myself ?

(Thanks for your patience, it's my first contribution)

Reply:
It's good that you didn't squash your commits. As you assumed, they will be squashed automatically when merged.

I don't know either why these many reviewers were requested...